### ISSUE TITLE
new paste with both raw and encrypted options shows the non-decrypted raw file

### ISSUE THREAD

**Issue by kit-ty-kate** [dinosaure/pasteur#14]
Tested on https://paste.osau.re/

**Comment by dinosaure** [comment]
Yes, actually the issue is more complex than what I through at the beginning. When you want to show up a paste in the "raw mode", we just response the content from the database. For obvious reason, we decrypt a paste only on the client side with JavaScript. In the case of the "raw mode", we don't run JavaScript at all (and we are not able to decrypt the content given by the server).

What is possible is to have a tool which "decrypt" the raw response (again, on the client side) and use it, like `wget https://paste... | pasteur-decrypt`.

**Comment by kit-ty-kate** [comment]
why can't we just do the decryption server side when in raw mode?

**Comment by dinosaure** [comment]
> why can't we just do the decryption server side when in raw mode?

The server does not have the key :smiling_face_with_tear:. The key is only generated and available on the client side.
